### ISSUE TITLE
Cannot reconnect. Please reload the window.

### ISSUE THREAD

**Issue by AtyKlaxas** [coder/code-server#4773]
## OS/Web Information

- Web Browser: Firefox (96.0.1 (64 bits)) / Google Chrome (Version 97.0.4692.71 (Build officiel) (arm64))
- Local OS: MacOS (12.1)
- Remote OS: Ubuntu 20.04.3 LTS
- Remote Architecture: x86-64
- `code-server --version`: `4.0.1 735c6da829535969ff7193c79379299e4a1cb9bc`

## Steps to Reproduce

1. Install via `curl -fsSL https://code-server.dev/install.sh | sh`
2. Start code-server
3. Go to the code server page
4. use code-server normally or wait for 1-2 min

## Expected

not the message `Cannot reconnect. Please reload the window.`

## Actual

the message `Cannot reconnect. Please reload the window.`

## Logs

Browser log : 
```
Échec du chargement de la carte source par les Outils de développement : Impossible de charger le contenu de https://<serverDomain>:8045/service-worker.js.map : Erreur HTTP : code d'état 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
log.ts:295 DEBUG [Extensions] Coder: Enabled All Proposed APIs
log.ts:301  INFO [Service Worker] registered
log.ts:295 DEBUG Checking for update...
log.ts:295 DEBUG Terminal process ready (shellProcessId: 277047)
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] resolving connection...
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] connecting to <serverDomain>:8045...
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] reconnected!
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] resolving connection...
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] connecting to <serverDomain>:8045...
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] reconnected!
log.ts:295 DEBUG Terminal process exit (instanceId: 1) with code undefined
log.ts:295 DEBUG Terminal process exit (instanceId: 1) state 5
log.ts:313   ERR Error received from starting extension host (kind: Remote)
log.ts:313   ERR The remote extenion host took longer than 60s to send its ready message.
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] resolving connection...
log.ts:301  INFO [remote-connection][ExtensionHost][ce7be…][reconnect] connecting to <serverDomain>:8045...
log.ts:313   ERR [remote-connection][ExtensionHost][ce7be…][reconnect][<serverDomain>:8045] received error control message when negotiating connection. Error:
log.ts:313   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (remoteAgentConnection.ts:732:17)
    at remoteAgentConnection.ts:327:17
    at l.fire (event.ts:577:16)
    at u.fire (ipc.net.ts:511:19)
    at r._receiveMessage (ipc.net.ts:873:27)
    at ipc.net.ts:818:73
    at l.fire (event.ts:577:16)
    at d.acceptChunk (ipc.net.ts:284:21)
    at ipc.net.ts:245:51
    at browserSocketFactory.ts:197:39
log.ts:313   ERR [remote-connection][ExtensionHost][ce7be…][reconnect] A permanent error occurred in the reconnecting loop! Will give up now! Error:
log.ts:313   ERR Error: Connection error: Unknown reconnection token (seen before)
    at W (remoteAgentConnection.ts:732:17)
    at remoteAgentConnection.ts:327:17
    at l.fire (event.ts:577:16)
    at u.fire (ipc.net.ts:511:19)
    at r._receiveMessage (ipc.net.ts:873:27)
    at ipc.net.ts:818:73
    at l.fire (event.ts:577:16)
    at d.acceptChunk (ipc.net.ts:284:21)
    at ipc.net.ts:245:51
    at browserSocketFactory.ts:197:39
abstractExtensionService.ts:644 Extension host (Remote) terminated unexpectedly. Code: 0, Signal: null
```

`code-server-stderr.log` : 
```
(...)
[10:54:06] [<clientIP>][ce7be672][ExtensionHostConnection] Unknown reconnection token (seen before).
```

```
(...)

[2022-01-20T09:38:26.829Z] debug 7 active connections
[2022-01-20T09:39:26.831Z] debug 6 active connections
[2022-01-20T09:40:26.833Z] debug 6 active connections
[2022-01-20T09:41:26.835Z] debug 6 active connections
[2022-01-20T09:42:26.838Z] debug 6 active connections
[2022-01-20T09:43:26.840Z] debug 6 active connections
[2022-01-20T09:44:26.841Z] debug 6 active connections
[2022-01-20T09:45:26.842Z] debug 6 active connections
[2022-01-20T09:46:26.844Z] debug 6 active connections
[2022-01-20T09:47:26.844Z] debug 6 active connections
[2022-01-20T09:48:26.846Z] debug 6 active connections
[2022-01-20T09:49:26.849Z] debug 6 active connections
[2022-01-20T09:50:26.853Z] debug 6 active connections
[2022-01-20T09:51:26.855Z] debug 6 active connections
[2022-01-20T09:52:26.857Z] debug 6 active connections
[10:53:04] [Theme Service] Attempting to find user's active theme: Default Dark+
[10:53:04] [Theme Service] Attempting to find user's active theme: Default Dark+
[10:53:04] [<clientIP>][c275705e][ManagementConnection] New connection established.
[2022-01-20T09:53:05.653Z] debug got latest version [38;2;140;140;140m{"latest":"4.0.1"}[0m
[2022-01-20T09:53:05.653Z] debug comparing versions [38;2;140;140;140m{"current":"4.0.1","latest":"4.0.1"}[0m
[10:53:05] [<clientIP>][ce7be672][ExtensionHostConnection] New connection established.
[10:53:05] [<clientIP>][ce7be672][ExtensionHostConnection] <277346> Launched Extension Host Process.
[10:53:06] [File Watcher (parcel)] Request to start watching: /home/<user> (excludes: **/.git/objects/**,**/.git/subtree-cache/**,**/node_modules/*/**,**/.hg/store/**,/home/<user>/.local/share/code-server/extensions/**)
[10:53:06] [File Watcher (parcel)] Request to stop watching: 
[10:53:06] [File Watcher (parcel)] Started watching: '/home/<user>' with backend 'inotify' and native excludes '/home/<user>/.git/objects, /home/<user>/.git/subtree-cache, /home/<user>/node_modules, /home/<user>/.hg/store, /home/<user>/.local/share/code-server/extensions'
[IPC Library: Pty Host]  INFO Persistent process "1": Replaying 115 chars and 1 size events
[10:53:26] [<clientIP>][ce7be672][ExtensionHostConnection] The client has reconnected.
[2022-01-20T09:53:26.859Z] debug 9 active connections
[10:53:46] [<clientIP>][ce7be672][ExtensionHostConnection] The client has reconnected.
[10:54:06] [<clientIP>][ce7be672][ExtensionHostConnection] <277346> Extension Host Process exited with code: 0, signal: null.
[10:54:06] [<clientIP>][c275705e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
[2022-01-20T09:54:26.861Z] debug 11 active connections
[2022-01-20T09:55:26.862Z] debug 10 active connections
[2022-01-20T09:56:26.864Z] debug 9 active connections
[2022-01-20T09:57:26.865Z] debug 9 active connections
[2022-01-20T09:58:26.866Z] debug 9 active connections
```

## Notes

This issue can be reproduced in VS Code: idk

**Comment by jsjoeio** [comment]
Duplicate of https://github.com/coder/code-server/issues/4693

**Comment by antofthy** [comment]
I am seeing this with code-server v4.0.2   
```
code-server: v4.0.2
VS Code: v1.63.0
Commit: 5cdfe74686aa73e023f8354a9a6014eb30caa7dd
Date: 2022-01-27T22:22:39Z (3 wks ago)
Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36
```

**Comment by jsjoeio** [comment]
@antofthy the fix for this is already in `main`. If you are using yarn/npm, you can use it today with `yarn global add code-server@beta` or you can use it when officially released in 4.0.3

Follow along [here](https://github.com/coder/code-server/milestone/36).
